File: src/main/java/com/letgo/book/domain/BookCreated.java
package com.letgo.book.domain;

import com.letgo.shared.domain.DomainEvent;

final public class BookCreated extends DomainEvent {
    private final String title;

    public BookCreated(String id, String title) {
        super(id);
        this.title = title;
    }

    public String title() {
        return title;
    }

    @Override
    public String name() {
        return "BookCreated";
    }

    @Override
    public String body() {
        return "{'title': " + title + "}";
    }
}

File: src/main/java/com/letgo/book/domain/BookCreated.kt
package com.letgo.book.domain

import com.letgo.shared.domain.DomainEvent
import java.time.LocalDateTime
import java.util.UUID

data class BookCreated(
    override val aggregateId: String,
    val title: String,
    override val occurredOn: LocalDateTime,
    override val id: UUID = UUID.randomUUID(),
) : DomainEvent
